hydra_provider/src/mavlink_service/service.rs

```diff
@@ -0,0 +1,41 @@
+use messages::mavlink::connect;
+use messages::mavlink::uorocketry::MavMessage;
+use messages::mavlink::MavConnection;
+use messages::mavlink::MavHeader;
+
+/**
+ * Handles all interactions with the Mavlink module
+ */
+pub struct MavlinkService {
+    pub mavlink: Option<Box<dyn MavConnection<MavMessage> + Send + Sync>>,
+    message_count: u8,
+}
+
+impl MavlinkService {
+    pub fn new() -> MavlinkService {
+        MavlinkService {
+            mavlink: None,
+            message_count: 0,
+        }
+    }
+
+    pub fn reconnect(&mut self, port: i32, baud_rate: i32) {
+        let address = format!("serial:{}:{}", port, baud_rate);
+        self.mavlink = Some(connect::<MavMessage>(address.as_str()).unwrap());
+    }
+
+    pub fn read_next(&self) -> MavMessage {
+        let (_, mavlink_message) = self.mavlink.as_ref().unwrap().recv().unwrap();
```

Comment:
Should we let the system crash if there is a failed message unwrap?

Reply:
hmm from the usage (L27)

https://github.com/uorocketry/rgs/blob/bea0660759ecc64f01c531755bb11a8e6852604a/hydra_provider/src/data_feed_service/serial/iterator.rs#L21-L46

I would think is best read_next returns an options or result. 
There are a lot of way things could fail (eg: signal lost, timeout, a corrupted message somehow)

Reply:
Yeah I'm gonna switch it to Result<>